### PR TITLE
chapter 7 practic

### PR DESCRIPTION
**Pull request by Dmitriy-Romanchuk**

### CODE REVIEW

File: chapter_7_practic/task1.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	std::cout << array;
+
+	return EXIT_SUCCESS;
+}
+
+void reversit(char str[])
+{
+	char temp = ' ';
+	
+	for (int i = 0; i < strlen(str) / 2; i++)
```
Comment (zefir-o) [inline chapter_7_practic/task1.cpp]: prefer prefix increment if possible, e.g.`++i`

Comment (zefir-o) [inline chapter_7_practic/task1.cpp]: In the body of the loop you call 3 times per iteration the function strlen(str). The result of it never changes, as the length of the given string is a constant.
The function underneath iterates over all characters and tries to find the '\0' char. If the length of the string is megabytes, it's very slow.
Prefer, if possible, call the function once before the loop and use the result of the function in the body-loop.

File: chapter_7_practic/task10.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+class matrix
+{
+private:
+	static const int max_1 = 10;
+	static const int max_2 = 10;
+	int limit[max_1][max_2]{};
+	int custom_max_1 = 0;
+	int custom_max_2 = 0;
+public:
+	matrix()
```
Comment (zefir-o) [inline chapter_7_practic/task10.cpp]: default constructor is not needed, thus should not be defined

File: chapter_7_practic/task10.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	{}
+	matrix(int a, int b)
```
Comment (zefir-o) [inline chapter_7_practic/task10.cpp]: do you know class' initializer list?

File: chapter_7_practic/task2.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	char repeat = 'y';
+	int counter = 0;
+
+	for (int j = 0; j < MAX; j++)
+	{
+		managers[j].getdata();
+		std::cout << "y - to the next name. Press any button to exit ";
+		repeat = getche();
+		std::cout << std::endl;
+		counter++;
```
Comment (zefir-o) [inline chapter_7_practic/task2.cpp:48]: prefer prefix increment

Comment (zefir-o) [reply chapter_7_practic/task2.cpp:48]: here and below

File: chapter_7_practic/task2.cpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+		if ('y' == repeat)
+			continue;
+		else             //output
+		{
+			for (int i = 0; i < counter; i++)
+			{
+				managers[i].putdata();
+			}
+			exit(EXIT_SUCCESS);
```
Comment (zefir-o) [inline chapter_7_practic/task2.cpp]: to stop the loop/iteration, here is special keyword: `break`

File: chapter_7_practic/task9.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	{
+		if (tail == 10 - 1)
+		{
+			tail = -1;
+		}
+		que[++tail] = var;
+	}
+	int get()
+	{
+		int temp = que[head++];
```
Comment (zefir-o) [inline chapter_7_practic/task9.cpp:44]: the function should remove an element from the queue

**Review by zefir-o — APPROVED** [review]

File: chapter_7_practic/task1.cpp (hunk 1 above)
Comment (Dmitriy-Romanchuk) [reply chapter_7_practic/task1.cpp]: https://github.com/Dmitriy-Romanchuk/master/pull/7/commits/2d419926d3f3d06a39d2a28077bc3d13c4af352b

Comment (Dmitriy-Romanchuk) [reply chapter_7_practic/task1.cpp]: https://github.com/Dmitriy-Romanchuk/master/pull/7/commits/2d419926d3f3d06a39d2a28077bc3d13c4af352b

File: chapter_7_practic/task10.cpp (hunk 3 above)
Comment (Dmitriy-Romanchuk) [reply chapter_7_practic/task10.cpp]: https://github.com/Dmitriy-Romanchuk/master/pull/7/commits/d6da9e7105927079cf4a81de56d1e60ab6ae37a7

File: chapter_7_practic/task10.cpp (hunk 2 above)
Comment (Dmitriy-Romanchuk) [reply chapter_7_practic/task10.cpp]: https://github.com/Dmitriy-Romanchuk/master/pull/7/commits/d6da9e7105927079cf4a81de56d1e60ab6ae37a7

File: chapter_7_practic/task2.cpp (hunk 5 above)
Comment (Dmitriy-Romanchuk) [reply chapter_7_practic/task2.cpp]: https://github.com/Dmitriy-Romanchuk/master/pull/7/commits/cb7996700791db4a4b2fc3e043fb5727032ea53b

File: chapter_7_practic/task2.cpp (hunk 4 above)
Comment (Dmitriy-Romanchuk) [reply chapter_7_practic/task2.cpp:48]: https://github.com/Dmitriy-Romanchuk/master/pull/7/commits/cb7996700791db4a4b2fc3e043fb5727032ea53b

**Pull request merged**